app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -51,31 +60,45 @@ def determine_os_type(host)
       os_name = os_name_mapping(host)
       arch_name = arch_name_mapping(host)
 
-      best_match = available_operating_systems.select { |os| os.name.present? }.max_by do |os|
+      best_match = available_operating_systems.select { |os| os[:name].present? }.max_by do |os|
         rating = 0.0
-        if os.name.include?(os_name)
+        if os[:name].include?(os_name)
           rating += 100
-          rating += (1.0/os.name.length) # prefer the shorter names a bit in case we have not found more important some specifics
-          rating += 10 if os.name.include?("#{os_name}_#{host.operatingsystem.major}")
-          rating += 10 if arch_name && os.name.include?(arch_name)
+          rating += (1.0/os[:name].length) # prefer the shorter names a bit in case we have not found more important some specifics
+          rating += 10 if os[:name].include?("#{os_name}_#{host.operatingsystem.major}")
+          rating += 10 if arch_name && os[:name].include?(arch_name)
         end
         rating
       end
 
-      best_match.name if best_match
+      best_match[:name] if best_match
     end
 
     def available_operating_systems
-      return @operating_systems if @operating_systems
-      if client.respond_to?(:operating_systems)
-        @operating_systems = client.operating_systems
-      elsif rbovirt_client.respond_to?(:operating_systems)
-        @operating_systmes = rbovirt_client.operating_systems
+      if attrs.key?(:available_operating_systems)
+        attrs[:available_operating_systems]
       else
         raise Foreman::Exception.new("Listing operating systems is not supported by the current version")
       end
     end
 
+    def update_available_operating_systems
+      ovirt_operating_systems = if client.respond_to?(:operating_systems)
+                                  client.operating_systems
+                                elsif rbovirt_client.respond_to?(:operating_systems)
+                                  rbovirt_client.operating_systems
```

Comment:
nit: I know you check it in `supports_operating_systems?` but since this is a public method, could you please add else branch raising some more meaningful error so it wouldn't fail NoMethodError nil#map? Or make this method private.